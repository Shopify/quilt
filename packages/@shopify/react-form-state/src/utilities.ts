import isEqual from 'fast-deep-equal';

export {isEqual};

export function mapObject<Input, Output>(
  input: Input,
  mapper: (value: any, key: keyof Input) => any,
) {
  return Object.entries(input).reduce((accumulator: any, [key, value]) => {
    accumulator[key] = mapper(value, key as keyof Input);
    return accumulator;
  }, {}) as Output;
}

export function push<T>(array: T[], ...values: T[]) {
  return array.concat(values);
}

export function remove<T>(array: T[], targetIndex: number) {
  return array.filter((_, index) => index !== targetIndex);
}

export function replace<T>(array: T[], targetIndex: number, newValue: T) {
  return array.map((value, index) => {
    if (index !== targetIndex) {
      return value;
    }

    return newValue;
  });
}

export function set<InputType extends object>(
  rootObject: InputType,
  path: string[],
  value: any,
) {
  if (path.length === 0) {
    return rootObject;
  } else if (path.length === 1) {
    return {
      ...(rootObject as any),
      [path[0]]: value,
    };
  } else {
    const [current, ...rest] = path;

    return {
      ...(rootObject as any),
      [current]: set(rootObject[current] || {}, rest, value),
    } as InputType;
  }
}

export function flatMap<T>(
  array: any[],
  mapper: (item: any, index?: number) => T | T[],
): T[] {
  return array.reduce(
    (acc, item, index) => acc.concat(mapper(item, index)),
    [],
  );
}
