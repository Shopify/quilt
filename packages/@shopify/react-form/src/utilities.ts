import {ChangeEvent} from 'react';

import get from 'get-value';

import {
  Validates,
  Validator,
  FieldOutput,
  FieldDictionary,
  Field,
  FormError,
} from './types';

export function isField<T extends object>(
  input: FieldOutput<T>,
): input is Field<T> {
  return (
    Object.prototype.hasOwnProperty.call(input, 'value') &&
    Object.prototype.hasOwnProperty.call(input, 'onChange') &&
    Object.prototype.hasOwnProperty.call(input, 'onBlur') &&
    Object.prototype.hasOwnProperty.call(input, 'defaultValue')
  );
}

export function mapObject<Output>(
  input: any,
  mapper: (value: any, key: any) => any,
) {
  return Object.keys(input).reduce((accumulator: any, key) => {
    const value = input[key];
    accumulator[key] = mapper(value, key);
    return accumulator;
  }, {}) as Output;
}

export function normalizeValidation<Value, Context extends object = {}>(
  input: Validates<Value, Context>,
): Validator<Value, Context>[] {
  return Array.isArray(input) ? input : [input];
}

export function isChangeEvent(
  value: any,
): value is ChangeEvent<HTMLInputElement> {
  return (
    typeof value === 'object' &&
    Reflect.has(value, 'target') &&
    Reflect.has(value.target, 'value')
  );
}

export function propagateErrors(
  fieldBag: {[key: string]: FieldOutput<any>},
  errors: FormError[],
) {
  errors.forEach(error => {
    if (error.field == null) {
      return;
    }

    const got = get(fieldBag, error.field);

    if (isField(got)) {
      if (got.error !== error.message) {
        got.setError(error.message);
      }
    }
  });
}

export function validateAll(fieldBag: {[key: string]: FieldOutput<any>}) {
  const fields = Object.values(fieldBag);
  const errors: FormError[] = [];

  function validate(field: Field<unknown>) {
    const message = field.runValidation();
    if (message) {
      errors.push({message});
    }
  }

  function validateDictionary(fields: FieldDictionary<any>) {
    Object.values(fields).forEach(validate);
  }

  for (const item of fields) {
    if (isField(item)) {
      validate(item);
    } else if (Array.isArray(item)) {
      item.map(validateDictionary);
    } else {
      validateDictionary(item);
    }
  }

  return errors;
}

export function noop() {}
