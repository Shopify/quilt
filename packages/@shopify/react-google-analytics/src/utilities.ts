// We want only the final segment of a domain, without any
// preceding subdomains. However, we do keep leading `.` when
// subdomains exist.
export function getRootDomain(domain: string) {
  const parts = domain.split('.');
  const baseDomainOnly = parts.slice(-2).join('.');

  return parts.length > 2 ? `.${baseDomainOnly}` : baseDomainOnly;
}

export function noop() {}
