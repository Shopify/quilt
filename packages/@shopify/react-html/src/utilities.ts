export const SERIALIZE_ATTRIBUTE = 'data-serialized-id';
export const MANAGED_ATTRIBUTE = 'data-react-html';

export function getSerializationsFromDocument() {
  const serializations = new Map<string, unknown>();

  if (typeof document === 'undefined') {
    return serializations;
  }

  for (const node of document.querySelectorAll(`[${SERIALIZE_ATTRIBUTE}]`)) {
    serializations.set(
      node.getAttribute(SERIALIZE_ATTRIBUTE)!,
      getSerializedFromNode(node),
    );
  }

  return serializations;
}

function getSerializedFromNode<Data>(node: Element): Data | undefined {
  const value = node.textContent;

  try {
    return value ? JSON.parse(value) : undefined;
  } catch {
    return undefined;
  }
}

export function getSerialized<Data>(id: string) {
  const node = document.querySelector(`[${SERIALIZE_ATTRIBUTE}="${id}"]`);

  if (node == null) {
    throw new Error(`No serializations found for id "${id}"`);
  }

  return getSerializedFromNode<Data>(node);
}

// We hide the document in development by default in order to prevent
// flashes of unstyled content. This function will show the document
// after the styles have loaded.
export function showPage(): Promise<void> {
  // eslint-disable-next-line no-process-env
  return process.env.NODE_ENV === 'development' &&
    typeof document !== 'undefined'
    ? new Promise(resolve => {
        setTimeout(() => {
          document.body.style.visibility = '';
          resolve();
        }, 0);
      })
    : Promise.resolve();
}
