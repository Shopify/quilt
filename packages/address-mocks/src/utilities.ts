import {fetch} from '@shopify/jest-dom-mocks';
import type {
  LoadCountriesResponse,
  LoadCountryResponse,
  ResponseError,
} from '@shopify/address-consts';
import {GRAPHQL_ENDPOINT} from '@shopify/address-consts';

import type {Fixtures, MockOptions} from './types';

const OPERATION_NAMES = ['countries', 'country'];

type FixtureLoader = () => Fixtures | Promise<Fixtures>;

export const defaultFixtureLoader: FixtureLoader = async () => {
  return (await import('./fixtures/fixtures-light')).default;
};

export function mockCountryRequestsLazy(
  loadFixtures: FixtureLoader = defaultFixtureLoader,
  options: MockOptions = {},
) {
  return fetch.post(requestMatcher, createMockHandler(loadFixtures), {
    overwriteRoutes: false,
    ...options,
  });
}

export function requestMatcher(url, options) {
  if (url !== GRAPHQL_ENDPOINT || typeof options.body !== 'string') {
    return false;
  }

  const body = JSON.parse(options.body);
  return OPERATION_NAMES.includes(body.operationName);
}

export function createMockHandler(loadFixtures: FixtureLoader) {
  let fixtures: Fixtures;
  return async function lazyRequestHandler(
    _,
    {body},
  ): Promise<LoadCountriesResponse | LoadCountryResponse | ResponseError> {
    if (!fixtures) {
      fixtures = await loadFixtures();
    }
    const locales = Object.keys(fixtures.countries);
    const {variables, operationName} = JSON.parse(body as string);
    const {countryCode, locale, includeHiddenZones} = variables || {};

    if (!locales.includes(locale)) {
      return {
        errors: [
          {
            message:
              'Variable $locale of type SupportedLocale! was provided invalid value',
          },
        ],
      };
    }

    const countries = fixtures.countries[locale];

    if (includeHiddenZones) {
      const indiaIndex = countries.data.countries.findIndex(
        (country) => country.code === 'IN',
      );

      // localized names of DH in the supported languages
      const dhNames = {
        // CLDR doesn't translate this name for AF, so use the EN fallback
        en: 'Dadra and Nagar Haveli and Daman and Diu',
        fr: 'Dadra et Nagar Haveli et Daman et Diu',
        ja: 'ダードラー及びナガル・ハヴェーリー及びダマン及びディーウ連邦直轄領',
      };

      countries.data.countries[indiaIndex].zones.push({
        name: dhNames[locale] || dhNames.en,
        code: 'DH',
      });
    }

    if (operationName === 'countries') return countries;

    const country = countries.data.countries.find(
      ({code}) => code === countryCode,
    );

    if (!country)
      return {
        errors: [
          {
            message:
              'Variable $countryCode of type SupportedCountry! was provided invalid value',
          },
        ],
      };
    return {
      data: {
        country,
      },
    };
  };
}
