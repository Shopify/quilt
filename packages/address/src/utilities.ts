import {Address, FieldName, Country, Zone} from './types';

const FIELD_REGEXP = /({\w+})/g;
export const FIELDS_MAPPING: {
  [key: string]: FieldName;
} = {
  '{firstName}': FieldName.FirstName,
  '{lastName}': FieldName.LastName,
  '{country}': FieldName.Country,
  '{city}': FieldName.City,
  '{zip}': FieldName.PostalCode,
  '{province}': FieldName.Zone,
  '{address1}': FieldName.Address1,
  '{address2}': FieldName.Address2,
  '{phone}': FieldName.Phone,
  '{company}': FieldName.Company,
};

/*
 * Returns empty string if all replacement fields are empty.
 */
export function renderLineTemplate(
  country: Country,
  template: string,
  address: Address,
): string {
  const result = template.match(FIELD_REGEXP);
  let line = template;

  if (!result) {
    return '';
  }

  let lineIsEmpty = true;
  result.forEach(key => {
    const addressKey = key.replace('{', '').replace('}', '') as FieldName;

    if (address[addressKey]) {
      lineIsEmpty = false;
    }

    switch (addressKey) {
      case FieldName.Country:
        line = line.replace(`{${FieldName.Country}}`, country.name);
        break;
      case FieldName.Zone:
        line = line.replace(
          `{${FieldName.Zone}}`,
          address.province ? getZone(country.zones, address.province).name : '',
        );
        break;
      default:
        line = line.replace(key, address[addressKey] || '');
        break;
    }
  });
  if (lineIsEmpty) {
    return '';
  } else {
    return line.trim().replace('  ', ' ');
  }
}

function getZone(zones: Zone[], zoneCode: string): Zone {
  return (
    zones.find(zone => zone.code === zoneCode) || {
      name: '',
      code: '',
    }
  );
}
