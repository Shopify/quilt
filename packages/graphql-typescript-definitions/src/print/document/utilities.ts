import {
  GraphQLCompositeType,
  // We need to bring these in as they are implicitly referenced by
  // GraphQLCompositeType, but TypeScript doesnâ€™t know this when it
  // generates its declaration files.
  // @ts-expect-error TS needs this implict reference though we never use it explicitly
  GraphQLInterfaceType,
  // @ts-expect-error TS needs this implict reference though we never use it explicitly
  GraphQLObjectType,
  // @ts-expect-error TS needs this implict reference though we never use it explicitly
  GraphQLUnionType,
} from 'graphql';
import {Field} from 'graphql-tool-utilities';

export class ObjectStack {
  private readonly seenFields = new Set<string>();

  get name(): string {
    const {parent, field, isFragment, type} = this;
    const fieldName = field ? upperCaseFirst(field.responseName) : '';
    const name = `${parent ? parent.name : ''}${fieldName}`;
    return isFragment ? `${name}${type ? type.name : 'Other'}` : name;
  }

  constructor(
    private readonly type?: GraphQLCompositeType,
    private readonly field?: Field,
    private readonly parent?: ObjectStack,
    private readonly isFragment = false,
  ) {}

  nested(field: Field, type: GraphQLCompositeType) {
    return new ObjectStack(type, field, this);
  }

  fragment(type?: GraphQLCompositeType) {
    return new ObjectStack(type, this.field, this.parent, true);
  }

  sawField(field: Field) {
    this.seenFields.add(field.responseName);
  }

  hasSeenField(field: Field) {
    return this.seenFields.has(field.responseName);
  }
}

export function upperCaseFirst(input: string) {
  return input.charAt(0).toUpperCase() + input.substr(1);
}
