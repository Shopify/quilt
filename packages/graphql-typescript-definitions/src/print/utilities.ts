import * as t from '@babel/types';
import {
  GraphQLString,
  GraphQLInt,
  GraphQLFloat,
  GraphQLBoolean,
  GraphQLID,
} from 'graphql';

export const scalarTypeMap = {
  [GraphQLString.name]: t.tsStringKeyword(),
  [GraphQLInt.name]: t.tsNumberKeyword(),
  [GraphQLFloat.name]: t.tsNumberKeyword(),
  [GraphQLBoolean.name]: t.tsBooleanKeyword(),
  [GraphQLID.name]: t.tsStringKeyword(),
};
