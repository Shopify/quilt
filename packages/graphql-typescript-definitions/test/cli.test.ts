/* eslint-disable @shopify/jest/no-snapshots */
import {resolve} from 'path';
import {exec} from 'child_process';

import {stripFullFilePaths} from '../../../tests/utilities';

const scriptPath = resolve(__dirname, '../bin/graphql-typescript-definitions');
const rootFixturePath = resolve(__dirname, 'fixtures');

// Skipping for now because it fails in CI for no apparent reason :(
// eslint-disable-next-line jest/no-disabled-tests
describe.skip('cli', () => {
  it('succeeds when there are no fixture errors', async () => {
    expect(
      await execDetails(cliCommandForFixtureDirectory('all-clear')),
    ).toMatchSnapshot();
  });

  it('succeeds when there are multiple schemas', async () => {
    expect(
      await execDetails(cliCommandForFixtureDirectory('multiple-schemas')),
    ).toMatchSnapshot();
  });

  it('succeeds when schemaPath is set to a graphql file', async () => {
    expect(
      await execDetails(cliCommandForFixtureDirectory('graphql-schema')),
    ).toMatchSnapshot();
  });

  it('fails when there are syntax errors', async () => {
    expect(
      await execDetails(cliCommandForFixtureDirectory('malformed-query')),
    ).toMatchSnapshot();
    expect(
      await execDetails(cliCommandForFixtureDirectory('missing-schema')),
    ).toMatchSnapshot();
  });

  it('fails when there are unused types', async () => {
    expect(
      await execDetails(cliCommandForFixtureDirectory('missing-types')),
    ).toMatchSnapshot();
  });

  it('fails when multiple queries have the same name', async () => {
    expect(
      await execDetails(cliCommandForFixtureDirectory('duplicate-query-names')),
    ).toMatchSnapshot();
  });

  it('fails when multiple fragments have the same name', async () => {
    expect(
      await execDetails(
        cliCommandForFixtureDirectory('duplicate-fragment-names'),
      ),
    ).toMatchSnapshot();
  });
});

function execDetails(command: string) {
  return new Promise((resolve) => {
    exec(command, (error, stdout, stderr) => {
      resolve({
        error: stripFullFilePaths(error),
        stdout: stripFullFilePaths(stdout),
        stderr: stripFullFilePaths(stderr),
      });
    });
  });
}

function cliCommandForFixtureDirectory(fixture: string) {
  const fixtureDirectory = resolve(rootFixturePath, fixture);
  return [
    scriptPath,
    `--cwd '${fixtureDirectory}'`,
    `--schema-types-path '${fixtureDirectory}'`,
  ].join(' ');
}

/* eslint-enable @shopify/jest/no-snapshots */
