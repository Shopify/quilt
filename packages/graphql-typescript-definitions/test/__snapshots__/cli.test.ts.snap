// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`cli fails when multiple fragments have the same name 1`] = `
Object {
  "error": Object {
    "cmd": "packages/graphql-typescript-definitions/bin/graphql-typescript-definitions --cwd 'packages/graphql-typescript-definitions/test/fixtures/duplicate-fragment-names' --schema-types-path 'packages/graphql-typescript-definitions/test/fixtures/duplicate-fragment-names'",
    "code": 1,
    "killed": false,
    "signal": null,
  },
  "stderr": "",
  "stdout": "
 BUILT  packages/graphql-typescript-definitions/test/fixtures/duplicate-fragment-names/schema.json â†’ packages/graphql-typescript-definitions/test/fixtures/duplicate-fragment-names/types

 ERROR  GraphQL fragments must have a unique name. The fragment personFields is declared in:
 packages/graphql-typescript-definitions/test/fixtures/duplicate-fragment-names/documents/Query1/FindPersonsQuery.graphql
 packages/graphql-typescript-definitions/test/fixtures/duplicate-fragment-names/documents/Query2/FindPersonQuery.graphql
Error: GraphQL fragments must have a unique name. The fragment personFields is declared in:
 packages/graphql-typescript-definitions/test/fixtures/duplicate-fragment-names/documents/Query1/FindPersonsQuery.graphql
 packages/graphql-typescript-definitions/test/fixtures/duplicate-fragment-names/documents/Query2/FindPersonQuery.graphql
    at duplicates.forEach (packages/graphql-typescript-definitions/lib/index.js)
    at Array.forEach (<anonymous>)
    at getDuplicateFragments.forEach (packages/graphql-typescript-definitions/lib/index.js)
    at Array.forEach (<anonymous>)
    at Builder.checkForDuplicateFragments (packages/graphql-typescript-definitions/lib/index.js)
    at Builder.<anonymous> (packages/graphql-typescript-definitions/lib/index.js)
    at Generator.next (<anonymous>)
    at packages/graphql-typescript-definitions/lib/index.js
    at new Promise (<anonymous>)
    at __awaiter (packages/graphql-typescript-definitions/lib/index.js)

",
}
`;

exports[`cli fails when multiple queries have the same name 1`] = `
Object {
  "error": Object {
    "cmd": "packages/graphql-typescript-definitions/bin/graphql-typescript-definitions --cwd 'packages/graphql-typescript-definitions/test/fixtures/duplicate-query-names' --schema-types-path 'packages/graphql-typescript-definitions/test/fixtures/duplicate-query-names'",
    "code": 1,
    "killed": false,
    "signal": null,
  },
  "stderr": "",
  "stdout": "
 BUILT  packages/graphql-typescript-definitions/test/fixtures/duplicate-query-names/schema.json â†’ packages/graphql-typescript-definitions/test/fixtures/duplicate-query-names/types

 ERROR  GraphQL operations must have a unique name. The operation FindPerson is declared in:
 packages/graphql-typescript-definitions/test/fixtures/duplicate-query-names/documents/Query1/FindPersonsQuery.graphql
 packages/graphql-typescript-definitions/test/fixtures/duplicate-query-names/documents/Query2/FindPersonQuery.graphql
Error: GraphQL operations must have a unique name. The operation FindPerson is declared in:
 packages/graphql-typescript-definitions/test/fixtures/duplicate-query-names/documents/Query1/FindPersonsQuery.graphql
 packages/graphql-typescript-definitions/test/fixtures/duplicate-query-names/documents/Query2/FindPersonQuery.graphql
    at duplicates.forEach (packages/graphql-typescript-definitions/lib/index.js)
    at Array.forEach (<anonymous>)
    at getDuplicateOperations.forEach (packages/graphql-typescript-definitions/lib/index.js)
    at Array.forEach (<anonymous>)
    at Builder.checkForDuplicateOperations (packages/graphql-typescript-definitions/lib/index.js)
    at Builder.<anonymous> (packages/graphql-typescript-definitions/lib/index.js)
    at Generator.next (<anonymous>)
    at packages/graphql-typescript-definitions/lib/index.js
    at new Promise (<anonymous>)
    at __awaiter (packages/graphql-typescript-definitions/lib/index.js)

",
}
`;

exports[`cli fails when there are syntax errors 1`] = `
Object {
  "error": Object {
    "cmd": "packages/graphql-typescript-definitions/bin/graphql-typescript-definitions --cwd 'packages/graphql-typescript-definitions/test/fixtures/malformed-query' --schema-types-path 'packages/graphql-typescript-definitions/test/fixtures/malformed-query'",
    "code": 1,
    "killed": false,
    "signal": null,
  },
  "stderr": "",
  "stdout": "
[7m[1m[32m BUILT [39m[22m[27m [2mpackages/graphql-typescript-definitions/test/fixtures/malformed-query/schema.json[22m â†’ packages/graphql-typescript-definitions/test/fixtures/malformed-query/types
[7m[1m[31m ERROR [39m[22m[27m Syntax Error: Expected {, found Name \\"name\\"
[2mGraphQLError: Syntax Error: Expected {, found Name \\"name\\"[22m
[2m    at syntaxError (node_modules/graphql/error/syntaxError.js)[22m
[2m    at expect (node_modules/graphql/language/parser.js)[22m
[2m    at many (node_modules/graphql/language/parser.js)[22m
[2m    at parseSelectionSet (node_modules/graphql/language/parser.js)[22m
[2m    at parseFragmentDefinition (node_modules/graphql/language/parser.js)[22m
[2m    at parseExecutableDefinition (node_modules/graphql/language/parser.js)[22m
[2m    at parseDefinition (node_modules/graphql/language/parser.js)[22m
[2m    at many (node_modules/graphql/language/parser.js)[22m
[2m    at parseDocument (node_modules/graphql/language/parser.js)[22m
[2m    at Object.parse (node_modules/graphql/language/parser.js)[22m

",
}
`;

exports[`cli fails when there are syntax errors 2`] = `
Object {
  "error": Object {
    "cmd": "packages/graphql-typescript-definitions/bin/graphql-typescript-definitions --cwd 'packages/graphql-typescript-definitions/test/fixtures/missing-schema' --schema-types-path 'packages/graphql-typescript-definitions/test/fixtures/missing-schema'",
    "code": 1,
    "killed": false,
    "signal": null,
  },
  "stderr": "",
  "stdout": "[7m[1m[31m ERROR [39m[22m[27m Schema not found. Expected to find the schema at 'packages/graphql-typescript-definitions/test/fixtures/missing-schema/schema.json' but the path does not exist. Check 'packages/graphql-typescript-definitions/test/fixtures/missing-schema/.graphqlconfig' and verify that schemaPath is configured correctly.
[2mError: Schema not found. Expected to find the schema at 'packages/graphql-typescript-definitions/test/fixtures/missing-schema/schema.json' but the path does not exist. Check 'packages/graphql-typescript-definitions/test/fixtures/missing-schema/.graphqlconfig' and verify that schemaPath is configured correctly.[22m
[2m    at Object.resolveSchemaPath (packages/graphql-tool-utilities/lib/utilities.js)[22m
[2m    at getGraphQLProjects.reduce (packages/graphql-tool-utilities/lib/config.js)[22m
[2m    at Array.reduce (<anonymous>)[22m
[2m    at Object.getGraphQLSchemaPaths (packages/graphql-tool-utilities/lib/config.js)[22m
[2m    at Builder.<anonymous> (packages/graphql-typescript-definitions/lib/index.js)[22m
[2m    at Generator.next (<anonymous>)[22m
[2m    at packages/graphql-typescript-definitions/lib/index.js[22m
[2m    at new Promise (<anonymous>)[22m
[2m    at __awaiter (packages/graphql-typescript-definitions/lib/index.js)[22m
[2m    at Builder.run (packages/graphql-typescript-definitions/lib/index.js)[22m

",
}
`;

exports[`cli fails when there are unused types 1`] = `
Object {
  "error": null,
  "stderr": "",
  "stdout": "
[7m[1m[32m BUILT [39m[22m[27m [2mpackages/graphql-typescript-definitions/test/fixtures/missing-types/schema.json[22m â†’ packages/graphql-typescript-definitions/test/fixtures/missing-types/types

[7m[1m[32m BUILT [39m[22m[27m [2mpackages/graphql-typescript-definitions/test/fixtures/missing-types/documents/Fragment.graphql[22m â†’ packages/graphql-typescript-definitions/test/fixtures/missing-types/documents/Fragment.graphql.d.ts
[7m[1m[32m BUILT [39m[22m[27m [2mpackages/graphql-typescript-definitions/test/fixtures/missing-types/documents/Query.graphql[22m â†’ packages/graphql-typescript-definitions/test/fixtures/missing-types/documents/Query.graphql.d.ts
",
}
`;

exports[`cli succeeds when schemaPath is set to a graphql file 1`] = `
Object {
  "error": null,
  "stderr": "",
  "stdout": "
[7m[1m[32m BUILT [39m[22m[27m [2mpackages/graphql-typescript-definitions/test/fixtures/graphql-schema/schema.graphql[22m â†’ packages/graphql-typescript-definitions/test/fixtures/graphql-schema/types

[7m[1m[32m BUILT [39m[22m[27m [2mpackages/graphql-typescript-definitions/test/fixtures/graphql-schema/documents/Query.graphql[22m â†’ packages/graphql-typescript-definitions/test/fixtures/graphql-schema/documents/Query.graphql.d.ts
",
}
`;

exports[`cli succeeds when there are multiple schemas 1`] = `
Object {
  "error": null,
  "stderr": "",
  "stdout": "
[7m[1m[32m BUILT [39m[22m[27m [2mpackages/graphql-typescript-definitions/test/fixtures/multiple-schemas/schema1/schema.json[22m â†’ packages/graphql-typescript-definitions/test/fixtures/multiple-schemas/schema1/types
[7m[1m[32m BUILT [39m[22m[27m [2mpackages/graphql-typescript-definitions/test/fixtures/multiple-schemas/schema2/schema.json[22m â†’ packages/graphql-typescript-definitions/test/fixtures/multiple-schemas/schema2/types

[7m[1m[32m BUILT [39m[22m[27m [2mpackages/graphql-typescript-definitions/test/fixtures/multiple-schemas/schema1/documents/Fragment.graphql[22m â†’ packages/graphql-typescript-definitions/test/fixtures/multiple-schemas/schema1/documents/Fragment.graphql.d.ts
[7m[1m[32m BUILT [39m[22m[27m [2mpackages/graphql-typescript-definitions/test/fixtures/multiple-schemas/schema1/documents/Query.graphql[22m â†’ packages/graphql-typescript-definitions/test/fixtures/multiple-schemas/schema1/documents/Query.graphql.d.ts
[7m[1m[32m BUILT [39m[22m[27m [2mpackages/graphql-typescript-definitions/test/fixtures/multiple-schemas/schema2/documents/Fragment.graphql[22m â†’ packages/graphql-typescript-definitions/test/fixtures/multiple-schemas/schema2/documents/Fragment.graphql.d.ts
[7m[1m[32m BUILT [39m[22m[27m [2mpackages/graphql-typescript-definitions/test/fixtures/multiple-schemas/schema2/documents/Query.graphql[22m â†’ packages/graphql-typescript-definitions/test/fixtures/multiple-schemas/schema2/documents/Query.graphql.d.ts
",
}
`;

exports[`cli succeeds when there are no fixture errors 1`] = `
Object {
  "error": null,
  "stderr": "",
  "stdout": "
[7m[1m[32m BUILT [39m[22m[27m [2mpackages/graphql-typescript-definitions/test/fixtures/all-clear/schema.json[22m â†’ packages/graphql-typescript-definitions/test/fixtures/all-clear/types

[7m[1m[32m BUILT [39m[22m[27m [2mpackages/graphql-typescript-definitions/test/fixtures/all-clear/documents/Fragment.graphql[22m â†’ packages/graphql-typescript-definitions/test/fixtures/all-clear/documents/Fragment.graphql.d.ts
[7m[1m[32m BUILT [39m[22m[27m [2mpackages/graphql-typescript-definitions/test/fixtures/all-clear/documents/Query.graphql[22m â†’ packages/graphql-typescript-definitions/test/fixtures/all-clear/documents/Query.graphql.d.ts
",
}
`;
